posthog/api/utils.py

```diff
@@ -46,7 +46,7 @@ def retrieve_entity_from(entity_id: Union[str, int], events: List[Dict], actions
         if event.get("id") == entity_id:
             return event
     for action in actions:
-        if action.get("id") == entity_id:
+        if str(action.get("id")) == entity_id:
```

Comment:
Q: Why was this change needed? It will create a conflict with https://github.com/PostHog/posthog/pull/7298

Reply:
@macobo I've updated with a [little comment](https://github.com/PostHog/posthog/pull/7398/commits/e67535174db6f0a32faa033b02f7055e94d27c2b) around the change, and added more restrictive typing

Reply:
@macobo actually, probably best to get this in now, as the conflicts will be easier to resolve the other way e.g. that way it's just some files should be removed.